### DISCUSSION TITLE
When click a tag, How the url jump? Which page to show the tag list?

### DISCUSSION THREAD

**Discussion by arashrun** [hexojs/hexo#5057] [Question / Answer]
Like this
`
<a href="<%- url_for(`/tags/${tag.name}`) %>">
`

How could i known which page the url used layout? I just want to custom the tag-list page

**Comment by yoshinorin** [comment]
@arashrun 

How about this?

```ejs
// EJS template
<% site.tags.sort('name').each((tag) => { %>
  <a href="<%- url_for('/tags/' + tag.slug) %>">
    <%= tag.name %>
  </a>
<% }); %>
```

```html
// Example of result
<a href="/tags/NET-Core">.NET Core </a>
<a href="/tags/AWS">AWS </a>
<a href="/tags/AWS-Aurora">AWS Aurora </a>
<a href="/tags/AWS-CDK">AWS CDK </a>
...
```

**Reply by arashrun** [reply]
Not actually. The problem is that when i click a tag，it can't jump to tag-list page。

```html
// A piece of code under tag page
<ul>
	<li class="menu-item">
	          <a href="/tags/hexo/">hexo(1)</a>
	</li>
	<li class="menu-item">
	          <a href="/tags/ejs/">ejs(1)</a>
	</li>
	<li class="menu-item">
	          <a href="/tags/node-js/">node.js(2)</a>
	</li>
	<li class="menu-item">
	          <a href="/tags/jquery/">jquery(1)</a>
	</li>
	<li class="menu-item">
	          <a href="/tags/vim/">vim(1)</a>
	</li>
	<li class="menu-item">
	          <a href="/tags/php/">php(1)</a>
	</li>
</ul>
```
For example, when i click `/tags/hexo/`, it can't get the right page

**Reply by yoshinorin** [reply]
@arashrun 

### 1. Install plugin

First you have to install `hexo-generator-tag` plugin. (I assume you have been already install it.)

> https://github.com/hexojs/hexo-generator-tag

### 2. create `tag.ejs` template

Create `tag.ejs`  templage in `<theme_dir>/layout` directory. You can use `tag-index.ejs, archive.ejs, index.ejs` instead of `tag.ejs`. Please see [hexo-generator-tag README](https://github.com/hexojs/hexo-generator-tag/tree/865864a56a0a57d8aab581b937c2a1728a53920f).

```
// example
<theme_dir>/layout/tag.ejs
<theme_dir>/layout/tag-index.ejs
```

### 3. Implement template

Implement `tag.ejs` template. Also you can include other template engine. Please see [hexo-theme-landscape](https://github.com/hexojs/hexo-theme-landscape/blob/7d126c085dbea0ceb64d5129b0cb3418f67b8389/layout/tag.ejs).

### 4. Run server

After run `hexo server` you can access `localhost:4000/tags/<tag_name>`.

---

Any questions?

**Reply by arashrun** [reply]
I know this steps. I did this before. But it's not work.

```bash
~npm list
+-- hexo-deployer-git@3.0.0
+-- hexo-generator-archive@1.0.0
+-- hexo-generator-category@1.0.0
+-- hexo-generator-index@2.0.0

+-- hexo-generator-tag@1.0.0

+-- hexo-renderer-ejs@1.0.0
+-- hexo-renderer-markdown-it@6.0.1
+-- hexo-renderer-stylus@2.0.1
+-- hexo-server@2.0.0
+-- hexo-theme-landscape@0.0.3
+-- hexo-toc@1.1.0
`-- hexo@5.4.1
```

[This is my tag.ejs](https://github.com/arashrun/self-hexo-theme/blob/master/layout/tag.ejs)

**Reply by yoshinorin** [reply]
@arashrun 

I submit a PR for example.

> https://github.com/arashrun/self-hexo-theme/pull/1

**Comment by lorezyra** [comment]
Have you created the `/source/tags/index.md` file? (This is from the root of the project and _not_ the theme folder...)

Inside that file you should have markdown like this:
```md
---
title: Tags
type: "tags"
layout: "tags"
---
```

Tags page shows the list of tags... the `/tag/{tag.name}/` page shows articles for that tag... The the `/tag/` page is generated by the [hexo-generator-tag](https://github.com/hexojs/hexo-generator-tag) plugin. As @yoshinorin  mentioned above, you need to specify how the tag data is layed out... So, you have to create the layout file for tags (list). The generator plugin only creates the data set for use in the layout files...
